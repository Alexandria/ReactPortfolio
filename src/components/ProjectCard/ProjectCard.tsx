import React from "react";
import "./ProjectCard.css";
import {
  Card,
  CardActionArea,
  CardMedia,
  CardContent,
  Typography,
  CardActions,
  Button
} from "@material-ui/core";
export interface Project {
  title: string;
  img: string;
  description: string;
  toolIcons: string[];
}

const ProjectCard: React.FC<Project> = (props: Project) => {
  const { title, img, description, toolIcons } = props;

  return (
    <Card>
      <CardActionArea>
        <CardMedia
          component="img"
          alt="Contemplative Reptile"
          height="140"
          image={img}
          title="Contemplative Reptile"
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            {title}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            {description}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        {toolIcons.map(icon => {
          return <img src={icon} height={40} width={40} alt={icon} />;
        })}
        <Button size="small" color="primary">
          Learn More
        </Button>
      </CardActions>
    </Card>
  );
};

export default ProjectCard;
