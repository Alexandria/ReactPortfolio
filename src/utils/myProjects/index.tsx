import { Project } from "../../components/ProjectCard/ProjectCard";
import typescript from "../../assests/icons/typescript.png";
import Meeplely from "../../assests/images/Meeplely.png";
import nodeJs from "../../assests/icons/nodeJs.png";
import react from "../../assests/icons/react.png";
import postgres from "../../assests/icons/postgres.png";
import python from "../../assests/icons/python.png";
import phpmyadmin from "../../assests/icons/phpmyadmin.png";
import insomnia from "../../assests/icons/insomnia.png";
import django from "../../assests/icons/django.jpg";

export const myProjects: Project[] = [
  {
    title: "Meeplely",
    img: Meeplely,
    description:
      "Full stack web application that will help you manage your board game collection.",
    toolIcons: [typescript, nodeJs, react, postgres, insomnia]
  },
  {
    title: "Justice for Barb Blog",
    img: Meeplely,
    description:
      "Full stack web application that will help you manage your board game collection.",
    toolIcons: [django, python, phpmyadmin]
  }
];

export const project: Project = {
  title: "Meeplely",
  img: Meeplely,
  description:
    "Full stack web application that will help you manage your board game collection.",
  toolIcons: [typescript, nodeJs, react, postgres]
};
