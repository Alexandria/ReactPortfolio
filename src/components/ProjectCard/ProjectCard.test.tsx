import React from "react";
import ProjectCard from "./ProjectCard";
import { shallowRender } from "../../utils/shallowRender";
import { project } from "../../utils/myProjects";

describe("ProjectCard", () => {
  it("renders expected structure", () => {
    const received = shallowRender(
      <ProjectCard
        title={project.title}
        description={project.description}
        toolIcons={project.toolIcons}
        img={project.img}
      />
    );
    expect(received).toMatchSnapshot();
  });
});
