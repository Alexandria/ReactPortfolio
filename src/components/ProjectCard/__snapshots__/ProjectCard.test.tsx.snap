// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectCard renders expected structure 1`] = `
<div>
  Project Card
</div>
`;
